ports/esp32/network_common.c

```diff
@@ -77,6 +77,8 @@ NORETURN void esp_exceptions_helper(esp_err_t e) {
             mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("Wifi Would Block"));
         case ESP_ERR_WIFI_NOT_CONNECT:
             mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("Wifi Not Connected"));
+        case ESP_ERR_NO_MEM:
+            mp_raise_msg(&mp_type_MemoryError, MP_ERROR_TEXT("WiFi Out of Memory"));
```

Comment:
MemoryError is usually related to a Python heap memory error.  I think it's worth keeping that distinction, so we can more easily diagnose "no Python heap left" from "no IDF heap left".

So I suggest making this an OSError with the same message, "Wifi Out of Memory".  That's still a lot better than "Wifi Unknown Error", and allows MemoryError to specifically relate to the Python GC heap.

Reply:
Fair point!  My take on this was that the average user may or may not be aware that there are actually two heaps; playing with WiFi is probably the second thing a user tries after blinking a LED :)  

Still, I also see the value in maintaining consistency with the rest of the errors in that switch case - I updated the PR accordingly.